### DISCUSSION TITLE
Quarto not compiling to Typst

### DISCUSSION THREAD

**Discussion by ptram** [quarto-dev/quarto-cli#6957] [Q&A] [answered]
### Description

Hi,

I'm trying to let Quarto (v1.4.376) generate a PDF file from the simplest of the documents. I use the `quarto typst compile [document path/name]` command, but get a series of errors that looks like Quarto is expecting a document using Typst syntax, not Pandoc. Hashtags, for example, are reported to have to preceed an expression.

The source document is very basic, and if you like you can see attached here.

Paolo
[quarto-typst-2023-09-20 (wrong).md](https://github.com/quarto-dev/quarto-cli/files/12704512/quarto-typst-2023-09-20.wrong.md)

**Comment by cscheid** [comment, marked as answer]
if you say "quarto typst", then you're invoking typst directly. If you want to use quarto's md to typst path, you need to type "quarto render" like other documents. The documentation may be helpful for you: https://quarto.org/docs/prerelease/1.4/typst.html

**Reply by ptram** [reply]
I see. However, now that I use the quarto render [document path/name] command, I still get errors. Quarto is asking for a document ('Safety.md') that is not indicated in the bare-bone file. This is the list of errors I get:

```
ERROR: Book chapter 'Safety.md' not found

Stack trace:
    at throwInputNotFound (file:///Applications/quarto/bin/quarto.js:88296:19)
    at findInputs (file:///Applications/quarto/bin/quarto.js:88327:17)
    at eventLoopTick (ext:core/01_core.js:181:11)
    at async findChapters (file:///Applications/quarto/bin/quarto.js:88339:13)
    at async bookRenderItems (file:///Applications/quarto/bin/quarto.js:88342:5)
    at async Object.bookProjectConfig [as config] (file:///Applications/quarto/bin/quarto.js:88248:25)
    at async projectContext (file:///Applications/quarto/bin/quarto.js:75151:37)
    at async render (file:///Applications/quarto/bin/quarto.js:80703:21)
    at async Command.fn (file:///Applications/quarto/bin/quarto.js:80904:32)
    at async Command.execute (file:///Applications/quarto/bin/quarto.js:8111:13)

```

Does it show some obvious mistake from my side? (I apologize for not being able to understand what the above means).

Paolo

**Reply by cscheid** [reply]
That's not the error i'm getting. I'm getting this:

```
ERROR: YAMLException: tab characters must not be used in indentation (quarto-typst-2023-09-20.wrong.md, 8:1)
7:     columns: 1
8: 	papersize: a5
  ~~
9: 	fontsize: 9.5
```

which indicates that your document has malformed YAML. Once you remove those lines (and fix the YAML that has other errors), then your file compiles fine.

````
---
title: "My Project"
format:
  typst:
    toc: true
    columns: 1
    papersize: a5
---

# This is a title

This is body text.
````

````
$ quarto render quarto-typst-2023-09-20.wrong.md
pandoc
  to: typst
  output-file: quarto-typst-2023-09-20.wrong.typ
  standalone: true
  default-image-extension: svg
  wrap: none
  citeproc: false
  toc: true

metadata
  columns: 1
  title: My Project
  papersize: a5

[typst]: Compiling quarto-typst-2023-09-20.wrong.typ to quarto-typst-2023-09-20.wrong.pdf...DONE

Output created: quarto-typst-2023-09-20.wrong.pdf
````

You should check that you have a sufficiently recent version of quarto and that you're not accidentally rendering the file in a quarto project (the _quarto.yml file might be in a parent folder).

**Reply by ptram** [reply]
> that you're not accidentally rendering the file in a quarto project (the _quarto.yml file might be in a parent folder).

This, in addition to other mistakes. Removing the bad lines clearly indicated by the compiler, I could generate my first Quarto PDF. Thank you!

Paolo

**Comment by ortizfr724** [comment]


**Reply by mcanouil** [reply]
@ortizfr724 please refrain reviving 3 years old thread with bad informations.
Providing bad information on purpose is a code of conduct violation.
